csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -134,22 +178,32 @@ std::unique_ptr<ReductionParams> getInnerOuterPersistentHeuristics(
         hp.threads_per_block_max,
         buffer_params.project_to_input,
         runtime_info.getIndexType());
-  } else {
-    inner_outer_multi_wave::getHeuristics(
-        rparams.get(),
-        properties.total_iteration_numel,
-        properties.total_reduction_numel,
-        buffer_params.regs_buffer_size,
-        buffer_params.smem_buffer_size,
-        buffer_params.smem_overhead,
-        max_outer_reduction_dtype_size,
-        hp.vectorize_factor,
-        hp.threads_per_block_min,
-        hp.threads_per_block_max,
-        buffer_params.project_to_input,
-        runtime_info.getIndexType());
+
+    // If warp specialized is enabled, or the heuristic is successful, return
+    if (hp.is_warp_specialized || rparams->is_good_ws_heuristic) {
```

Comment:
out of curiosity, what happens when we have `hp.is_warp_specialized` set as true, but we failed to get a good heuristics?